### DISCUSSION TITLE
how to re trigger re interview nodes

### DISCUSSION THREAD

**Discussion by akamming** [zwave-js/zwave-js#6593] [Q&A] [answered]
I just moved from openzwave to zwave js ui.  During the migration  i have 2 nodes i cannot get to work properly.

One of them is a neo coolcam wallplug. i couldn't find this wallplug when i did the migration. So the status now is Dead (and interview stage is protocolinfo). However when i try to activate it there is now way i can get the interviewing to finish. I did try 
- CLick the Node
- Click advanced
- CLikck reinterview node (and then instantly i get "API Refresh info ended succesfully"). In the "events" is also immediately states "interverview stage completed" and "interview failed"

The other device is a Heiman smoke detector. This one did finish the interview stage (after waking up, cause it's a battery powered device), but only exported the "battery is low" switch to mqtt, and not the other properties.  So  i wanted to reinterview (node/advanced/reinterview node). But also on this one i immediately get the "API Refresh info ended succesfully". But then nothing happens, so no "interview stage completed" and no "interview failed". Actually nothing happents in the tab events.

I have more Heiman smoke detector and Neo Coolcam wallplugs which all work correctly, so something really seemed to be going wrong during interviewing stage.

How can i trigger a proper re-interview?

**Comment by robertsLando** [comment]
Please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug` and attach it here as a file (drag & drop into the text field).

**Comment by akamming** [comment]
Tx that you are willing to take a look.

here's the log file (it is about node 185: The smoke sensor).
[debug.zip](https://github.com/zwave-js/node-zwave-js/files/13857886/debug.zip)

It is very weird, cause i have different behaviour now: the interviewing finished. 

However it is still nog complete, cause i am still missing discovery messages e.g. 
These are the discovery messages of the type binary sensor of well initialized smoke detector (the exact same device)
![image](https://github.com/zwave-js/node-zwave-js/assets/30364409/e888981c-bf16-44bb-aff7-88988d57d1e6)

but this failing one only shows
![image](https://github.com/zwave-js/node-zwave-js/assets/30364409/7df2e3c7-9fac-4568-a931-2debe52a83a5)

Also the discovery message for the battery level is missing.

You can forget about the Neo Coolcam i mentioned above. I did a factory reset and it turns out that node is dead, so no way to do representative testing on that one...

**Reply by robertsLando** [reply]
@AlCalzone Is the maintainer of the driver, please be patinet until he can take some time to review your logs.

For what I can see seems your node 185 suports S2 but you have no key setup on settings:

> 2024-01-08T08:31:35.344Z CNTRLR   [Node 185] supports Security S2, but no S2 network keys were configured. The i
                                  nterview might not include all functionality.

Try to add the S2 keys on settings, exclude and include the node again to see if the issues persists

**Comment by akamming** [comment]
tx.. i will try the reinclude with S2 security. Although i do not think this will have impact. The other Heiman smoke detecters in my network do finish correctly (also without S2)

**Reply by AlCalzone** [reply]
If it was definitely included without S2, there's no need to change that. That log message is more of a hint/warning.
I'll take a look at the logs when I find the time.

**Comment by akamming** [comment]
it was indeed included without s2. Tx for the help. i'll be patient ;-)

**Comment by AlCalzone** [comment, marked as answer]
Node 185 does not support Binary Sensor CC, but it supports Notification CC. You should have a notification value for *Smoke Alarm[Sensor status]*.

**Comment by akamming** [comment]
tx for investigating. I see the indeed no binary sensor in the ui. I now checked, cause i have several of these smoke detectors, all on the same hardware (Heiman). But i now checked. Not all have the same firmware. The ones with fw 1.1 have binary sensor support. The ones with more recent firmware (3.1.1) do not...

so it's indeed a thing in the device. Sorry to mention this as an issue on zwave js ui.  (and tx for the support!) 

BTW: I switched from openzwave to zwave js ui. It was quite a hassle to make it work, but now that it's working it's a big step forward. Much more options in the ui and the network is more stable and responsive. Tx a lot for all the hard work!!

**Reply by robertsLando** [reply]
Out of curiosity what has been the most difficult part in configuring Z-UI for you? Did you checked the official [docs](https://zwave-js.github.io/zwave-js-ui/#/)?

You feedback could help me improving them if something is not clear

**Reply by akamming** [reply]
I think the docs are quote clear and so is the product itself

most difficult part in moving from openzwave to zwavejsui for me was
- getting the containers for Mqtt and zwavejsui up and running. Openzwave was integrated into domoticz and I lacked docker and Mqtt knowledge. E.g. I spend a long time before I realised i Neede to configure that actual ip adress for the mqtt server instead of localhost.
- Getting the interview phase to be finished. I tried migration over a year ago and not all devices initialized. Now a year later they all did. So I assume Several bugs were fixed and that is why it’s working now

**Reply by robertsLando** [reply]
About the first bug there are many different ways to setup an mqtt broker container but I could integrate that in the docker setup.

About the second part the reasons could be multiple. It could be there is no config file in db for those devices yet, the device has some compatibilty problem, connectivity is bad (usually this can be fixed by using an usb extension cord), driver issue.

**Reply by akamming** [reply]
although never say never, but i think for 2nd networking or uncompatibility issues to be the cause is quite unlikely.  When i started doing the migration again at first i just started up the docker with just docker compose -d. (So basically recreated the container based on the release of a year ago) and it gave me exactly the same result (interview stage not finished, lots of device not recognized).

but then i did a docker compose down, docker compose pull and docker compose up -d again (so recreate image on latest releases version) and then it all worked...

anyway a year ago (which is a very rough estimate, i don't know exactly. It was a long time ago)  I did not spend a lot of time on diagnosing. I just checked if it worked because of some remarks of OpenZwave no more being maintained, but since the conversion gave me problems and domoticz still supported it at that time i just left it for a later time... 

And very happy to see it work now! and also much more stable and responsive than openZwave with better ui to manage the zwave network, so big step forward..!!!

**Reply by robertsLando** [reply]
Thanks for your feedback 🙏🏼
